### PR TITLE
Loc

### PR DESCRIPTION
**Pull request by snopoke**
http://manage.dimagi.com/default.asp?187497 + a bit of cleanup

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/cleanup.py [hunk 1]
```diff
@@ -62,7 +59,6 @@ def rebuild_case_from_forms(domain, case_id, detail):
     return FormProcessorInterface(domain).hard_rebuild_case(case_id, detail)
 
 
-@unit_testing_only
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/case/cleanup.py]: i did think this was good idea. maybe we could add `i_know_what_im_doing=True` as an option to override it in non-testing code

**Comment by czue** [comment]
comment was just thinking out loud, :+1: when tests pass

File: corehq/ex-submodules/casexml/apps/case/cleanup.py (hunk 1 above)
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/case/cleanup.py]: yea, I did think some additional check might be in order

On 20 November 2015 at 11:23, Cory Zue <notifications@github.com> wrote:

> In corehq/ex-submodules/casexml/apps/case/cleanup.py
> <https://github.com/dimagi/commcare-hq/pull/9198#discussion_r45449296>:
>
> > @@ -62,7 +59,6 @@ def rebuild_case_from_forms(domain, case_id, detail):
> >      return FormProcessorInterface(domain).hard_rebuild_case(case_id, detail)
> >
> >
> > -@unit_testing_only
>
> i did think this was good idea. maybe we could add
> i_know_what_im_doing=True as an option to override it in non-testing code
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/9198/files#r45449296>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

Comment (proteusvacuum) [inline corehq/ex-submodules/casexml/apps/case/cleanup.py]: lol

**Comment by czue** [comment]
build now failing
```
======================================================================
ERROR: test_update_status_set_location (corehq.apps.users.tests.fixture_status.TestFixtureStatus)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/users/tests/fixture_status.py", line 62, in test_update_status_set_location
    self.couch_user.set_location(fake_location)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/users/models.py", line 1807, in set_location
    location.group_id
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/jsonobject/utils.py", line 23, in update
    self[key] = value
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/jsonobject/containers.py", line 137, in __setitem__
    wrapped, unwrapped = self.__unwrap(key, value)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/jsonobject/containers.py", line 131, in __unwrap
    return self._wrapper.unwrap(wrapped)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/jsonobject/base_properties.py", line 218, in unwrap
    property_ = self.value_to_property(obj)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/jsonobject/base_properties.py", line 238, in value_to_property
    value, map_types_properties.keys())
BadValueError: value <MagicMock name='mock.group_id' id='140439430674896'> not in allowed types: [<type 'list'>, <type 'int'>, <type 'long'>, <type 'str'>, <type 'datetime.time'>, <type 'unicode'>, <class 'decimal.Decimal'>, <type 'bool'>, <type 'datetime.datetime'>, <type 'float'>, <type 'dict'>, <type 'datetime.date'>, <type 'set'>]
```

**Comment by snopoke** [comment]
:white_check_mark:

File: corehq/ex-submodules/casexml/apps/case/cleanup.py [hunk 2]
```diff
@@ -73,6 +72,12 @@ def safe_hard_delete(case):
 
     This is used primarily for cleaning up system cases/actions (e.g. the location delegate case).
     """
+    if not settings.UNIT_TESTING:
+        from corehq.apps.commtrack.const import USER_LOCATION_OWNER_MAP_TYPE
+        whitelist_case_types = [USER_LOCATION_OWNER_MAP_TYPE]
+        if case.type not in whitelist_case_types:
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/case/cleanup.py]: Is there a plan to add more types to this list? If not, how about `case.type == USER_LOCATION_OWNER_MAP_TYPE`?

File: corehq/ex-submodules/casexml/apps/case/cleanup.py (hunk 1 above)
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/case/cleanup.py]: The fact that there are unit-testing-only functions mixed with production code is a bit disconcerting.

File: corehq/ex-submodules/casexml/apps/case/cleanup.py (hunk 2 above)
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/case/cleanup.py]: Don't plan to add more types but this just felt like the intention was more
clear. Suppose a comment would also be fine.

On 20 November 2015 at 16:47, Daniel Miller <notifications@github.com>
wrote:

> In corehq/ex-submodules/casexml/apps/case/cleanup.py
> <https://github.com/dimagi/commcare-hq/pull/9198#discussion_r45475735>:
>
> > @@ -73,6 +72,12 @@ def safe_hard_delete(case):
> >
> >      This is used primarily for cleaning up system cases/actions (e.g. the location delegate case).
> >      """
> > +    if not settings.UNIT_TESTING:
> > +        from corehq.apps.commtrack.const import USER_LOCATION_OWNER_MAP_TYPE
> > +        whitelist_case_types = [USER_LOCATION_OWNER_MAP_TYPE]
> > +        if case.type not in whitelist_case_types:
>
> Is there a plan to add more types to this list? If not, how about case.type
> == USER_LOCATION_OWNER_MAP_TYPE?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/9198/files#r45475735>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

**Pull request merged**